src/webview/mod.rs

```diff
@@ -325,33 +299,18 @@ impl WebView {
     WebViewBuilder::new(window)?.build()
   }
 
-  /// Dispatch javascript code to be evaluated later. Note this will not actually run the
-  /// scripts being dispatched. Users need to call [`WebView::evaluate_script`] to execute them.
-  pub fn dispatch_script(&mut self, js: &str) -> Result<()> {
-    self.tx.send(js.to_string())?;
-    Ok(())
-  }
-
-  /// Create a [`Dispatcher`] to send evaluation scripts to the WebView. [`WebView`] is not thread
-  /// safe because it must be run on the main thread who creates it. [`Dispatcher`] can let you
-  /// send the scripts from other threads.
-  pub fn dispatcher(&self) -> Dispatcher {
-    Dispatcher(self.tx.clone())
-  }
-
   /// Get the [`Window`] associate with the [`WebView`]. This can let you perform window related
   /// actions.
   pub fn window(&self) -> &Window {
     &self.window
   }
 
-  /// Evaluate the scripts sent from [`Dispatcher`]s.
-  pub fn evaluate_script(&self) -> Result<()> {
-    while let Ok(js) = self.rx.try_recv() {
-      self.webview.eval(&js)?;
-    }
-
-    Ok(())
+  /// Evaluate and run javascript code. Must be called on the same thread who created the
+  /// [`WebView`]. Use [`EventLoopProxy`] and a custom event to send scripts from other threads.
+  ///
+  /// [`EventLoopProxy`]: crate::application::event_loop::EventLoopProxy
+  pub fn eval(&self, js: &str) -> Result<()> {
```

Comment:
The name of `eval` is more because of historical reasons.
Could you name it `evaluate_script`?

Reply:
OK. I closed #289 in favor of this PR, made the necessary adjustments in my draft PR [tauri-apps/tauri#1965](https://github.com/tauri-apps/tauri/pull/1965) and removed the draft status from this PR.